src/AWS/Lambda/Combinators.hs

```diff
@@ -70,3 +71,6 @@ withoutContext = const
 -- on a failure to convert the JSON AST sent to the Lambda.
 withInfallibleParse :: FromJSON a => (a -> b) -> Value -> b
 withInfallibleParse fn = either error fn . parseEither parseJSON
+
+withInfallibleParseEither :: FromJSON a => (a -> b) -> Value -> Either String b
+withInfallibleParseEither fn = fmap fn . parseEither parseJSON
```

Comment:
i think this function is below the [Fairbairn Threshold](https://wiki.haskell.org/Fairbairn_threshold), but if you want to keep it, it should be renamed. I'd call it `withFallibleParse` or `withFallibleParseEither`, because this parse is allowed to fail.